rwengine/src/engine/GameWorld.cpp

```diff
@@ -946,3 +958,57 @@ bool GameWorld::isPaused() const
 	return paused;
 }
 
+VehicleObject* GameWorld::tryToSpawnVehicle(VehicleGenerator& gen)
+{
+	constexpr float kMinClearRadius = 10.f;
+
+	bool clear = true;
+
+	if (gen.remainingSpawns <= 0) {
+		clear = false;
+	}
+
+	if (gen.lastSpawnTime + gen.minDelay > state->basic.timeMS) {
+		clear = false;
+	}
```

Comment:
while `clear` is a good name to check if the area is `clear` I feel like this (and the instance just before) is abuse of the variable. Maybe just `return nullptr` here or call the variable differently?

Early-out feels like a good idea in general for this function.